treadmill-scripts/newldap.sh

```diff
@@ -0,0 +1,184 @@
+    echo "Usage: $0 [option...]" >&2
+    echo
+    echo "   -s, --subnet           AWS subnet IDs; e.g. -s subnet-1234 -s subnet-5678 -s subnet-9999"
+    echo "   -a, --ami              AWS AMI image; e.g. ami-1234"
+    echo "   -c, --cell             New Treadmill cell name; e.g. userdev"
+    echo "   -d, --domain           IPA domain; e.g. ipa.foo.com"
+    echo "   -n, --number           Number of LDAP servers, e.g. 2"
+    echo "   -l, --ldapsuffix       LDAP suffix; e.g. dc=foo,dc=com"
+    echo "   -r, --realm            Realm; e.g. FOO.COM"
+    echo "   -p, --proid            LDAP auth user; e.g admin"
+    echo
+    exit 0
+}
+SUBNETS=()
+NUMBER=0
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -s|--subnet)
+    SUBNETS=("${SUBNETS[@]}" "$2")
+    shift # past argument
+    shift # past value
+    ;;
+    -a|--ami)
+    AMI="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -c|--cell)
+    TREADMILL_CELL="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -d|--domain)
+    TREADMILL_DNS_DOMAIN="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -n|--number)
+    NUMBER="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -l|--ldapsuffix)
+    TREADMILL_LDAP_SUFFIX="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -r|--realm)
+    REALM="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -p|--proid)
+    REALM="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    -h|--help)
+    display_help
+    ;;
+esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [ ! ${#SUBNETS[@]} -eq $NUMBER ]; then
+	echo -e "Please pass exactly the same number of subnets as requested number of servers; exiting."
+	exit 1
+fi
+
+echo 
+echo "AWS SUBNETS       = ${SUBNETS[@]}"
+echo "AWS AMI           = ${AMI}"
+echo "TREADMILL CELL    = ${TREADMILL_CELL}"
+echo "TREADMILL DOMAIN  = ${TREADMILL_DNS_DOMAIN}"
+echo "COUNT             = ${NUMBER}"
+echo "LDAP SUFFIX       = ${TREADMILL_LDAP_SUFFIX}"
+echo "REALM             = ${REALM}"
+echo "PROID             = ${TREADMILL_PROID}"
+echo
+
+echo "==="
+while true
+do
+	read -p "Continue (y/n)? " answer
+	case $answer in
+		[yY]* ) break;;
+		[nN]* ) echo "Exiting."; exit 1;;
+		* )     echo "Invalid input";;
+	esac
+done
+
+# Insert null value at pos 0 to index array from 1
+SUBNETS=("" "${SUBNETS[@]}")
+
+# Generate tmpdir for cell definitions
+tmpdir=$(mktemp -d -p /tmp)
+echo "Created temp working directory ${tmpdir}"
+if [[ ! "${tmpdir}" || ! -d "${tmpdir}" ]]; then
+	echo "Could not create temp dir"
+	exit 1
+fi
+
+# Deletes the temp directory on exit
+function cleanup {      
+	rm -rf "${tmpdir}"
+	echo "Deleted temp working directory ${tmpdir}"
+}
+trap cleanup EXIT
+
+# Generate list of LDAP masters
+TREADMILL_LDAP=''
+for i in `seq 1 ${NUMBER}`
+do
+	TREADMILL_LDAP+=ldap://${TREADMILL_CELL}-ldap-${i}.${TREADMILL_DNS_DOMAIN}:22389,
```

Comment:
see my prev comment - ldap is cross cell, so having <cell> in the name is not proper and not needed.

Reply:
Removed